tiledb/sm/filter/filter_create.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by EH on 11/21/2021.
+//
+
+#ifndef TILEDB_FILTER_CREATE_H
+#define TILEDB_FILTER_CREATE_H
+
+#include "filter.h"
+
+namespace tiledb::sm {
+
+class FilterCreate {
+ public:
+  /**
+   * Factory method to make a new Filter instance of the given type.
+   *
+   * @param type Filter type to make
+   * @return New Filter instance or nullptr on error.
```

Comment:
Just to be consistent and as informative as deserialize

```suggestion
   * @return New Filter instance (caller's responsibility to free) or nullptr on error.
```